contrib/fine-tuning/fine-tune-mlx.sh

```diff
@@ -38,11 +47,19 @@ node validator.js ${DATASET_PATH}
 
 # This step always pulls the latest base model from HF. Need to think about versioning and checksum to prevent model injection attacks
 echo "Test base model with the prompt 'Tell me about cdxgen'. Usually yields a low-quality response."
-mlx_lm.generate --model ${BASE_MODEL} --prompt "Tell me about cdxgen" --temp 0.05 --max-tokens 32000
+mlx_lm.generate --model ${BASE_MODEL} --prompt "Tell me about cdxgen" --temp 0.7 --max-tokens ${MAX_TOKENS}
 
 # We use LoRA fine-tuning over DoRA due to better compatibility with vLLM and llama.cpp
-echo "Low-Rank Adaptation (LoRA) fine-tuning ${BASE_MODEL} with cdx1 dataset. This might take a while ..."
-mlx_lm.lora --model ${BASE_MODEL} --train --data ${DATASET_PATH} --adapter-path ${ADAPTERS_PATH} --mask-prompt --fine-tune-type lora --batch-size 1 --num-layers ${NUM_LAYERS} --iters 2000 --grad-checkpoint --max-seq-length 16000 --learning-rate "3e-5" --optimizer adam
+if [ "$TOOL_BASE_MODEL" = "cdx1-mini" ]; then
+  echo "Full fine-tune with cdx-docs dataset. This might take a while ..."
+  mlx_lm.lora --model ${BASE_MODEL} --train --data ${DATASET_PATH} --adapter-path ${ADAPTERS_PATH} --mask-prompt --fine-tune-type full --batch-size 2 --num-layers ${NUM_LAYERS} --iters ${ITERS} --grad-checkpoint --max-seq-length ${MAX_SEQ} --learning-rate "1e-5" --optimizer adamw
+elif [ "$TOOL_BASE_MODEL" = "cdx1" ]; then
+  echo "Low-Rank Adaptation (LoRA) fine-tuning ${BASE_MODEL} with cdx-docs dataset. This might take a while ..."
+  mlx_lm.lora --model ${BASE_MODEL} --train --data ${DATASET_PATH} --adapter-path ${ADAPTERS_PATH} --mask-prompt --fine-tune-type lora --batch-size 1 --num-layers ${NUM_LAYERS} --iters ${ITERS} --grad-checkpoint --max-seq-length ${MAX_SEQ} --learning-rate "1e-4" --optimizer adamw
```

Comment:
I have started a new fine-tuning job for `cdx1`. Will create a separate PR tomorrow once it is ready for reevaluation.